File: backend/internal/models/post.go
package models

import "time"

type Post struct {
    ID        int64      `json:"id"`
    ModuleID  int64      `json:"module_id"`

    UserID    int64      `json:"user_id"`
    Username  string     `json:"username"`

    Title     string     `json:"title"`
    Content   string     `json:"content"`
    
    IsDeleted bool       `json:"is_deleted"`
    CreatedAt time.Time  `json:"created_at"`
    UpdatedAt *time.Time `json:"updated_at"`
}

File: frontend/src/types/post.ts
export interface Post {
  id: number;
  module_id: number;
  user_id: number;
  title: string;
  content: string;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
  username?: string; // Often joined from the users table in the backend
}
